server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java

```diff
@@ -63,8 +70,23 @@ public DialectService(
    */
   public ResultWithErrors<DialectOutcome> process(
       List<String> dialects, DialectProcessingContext context) {
-    List<CobolDialect> orderedDialects = sortDialects(dialects);
     List<SyntaxError> errors = new LinkedList<>();
+    List<CobolDialect> orderedDialects;
+    try {
+      orderedDialects = sortDialects(dialects);
+    } catch (NoSuchElementException e) {
+      Optional<String> originalDialectName = findOriginalDialect(e.getMessage());
```

Comment:
Too big a "catch" section. Can this functionality be moved to separate private function that returns the `ResultWithErrors` object?